src/js/components/NewConnectionModal.tsx

```diff
@@ -110,22 +111,25 @@ export default function NewConnectionModal() {
           return obj
         }, {})
         try {
+          setIsFetching(true)
           await dispatch(setConnection(toCluster({host})))
         } catch (_) {
           setErrors([{message: "Cannot connect to host"}])
           return
+        } finally {
+          setIsFetching(false)
         }
         closeModal()
       } else {
         setErrors(form.getErrors())
       }
     },
-    [f, config]
+    [f, config, setIsFetching]
```

Comment:
I don't think setIsFetching is necessary to include the deps here. I think the setters won't get stale like the values will.